win32/src/threadpool_win32.c

```diff
@@ -367,18 +286,9 @@ int threadpool_schedule_work_item(THANDLE(THREADPOOL) threadpool, THREADPOOL_WOR
         THREADPOOL* threadpool_ptr = THANDLE_GET_T(THREADPOOL)(threadpool);
 
         (void)InterlockedIncrement(&threadpool_ptr->pending_api_calls);
-
-        THREADPOOL_WIN32_STATE state = InterlockedAdd(&threadpool_ptr->state, 0);
-        if (state != THREADPOOL_WIN32_STATE_OPEN)
-        {
-            LogWarning("Bad state: %" PRI_MU_ENUM, MU_ENUM_VALUE(THREADPOOL_WIN32_STATE, state));
-        }
-        else
-        {
-            /* Codes_SRS_THREADPOOL_WIN32_05_013: [ threadpool_schedule_work_item shall call SubmitThreadpoolWork to submit the work item for execution. ]*/
-            SubmitThreadpoolWork(work_item_context->ptp_work);
-            result = 0;
-        }
+        /* Codes_SRS_THREADPOOL_WIN32_05_013: [ threadpool_schedule_work_item shall call SubmitThreadpoolWork to submit the work item for execution. ]*/
+        SubmitThreadpoolWork(work_item_context->ptp_work);
+        result = 0;
         (void)InterlockedDecrement(&threadpool_ptr->pending_api_calls);

```

Comment:
We usually have this be the last thing we do (between ln 293 and 294).

That's easier to spot usually than if it is in the middle of other code. /consider.

Reply:
It is matching the Increment on line 288. I am unsure how much below it can go.

Reply:
The comment in iteration 4 refers to " result = 0;" not to the InterlockedDecrement.

Reply:
Okay, but in the previous code, the result = 0 happened before the InterlockedDecrement. I am not sure how does it matter?

Reply:
Fair enough, let me explain why it matters: setting the result in the middle of the code of a function and then doing more things is prone to being overlooked when reviewing. Thus as a convention for our team we set the result as the last thing (unless there is a reason not to do that).

This is not such a case, there is no functional reason why it has to be done before the decrement.
So, it should be as the last thing we do, as everyone in the team expects it.

A question for you is: what technical reason is there to have the assignment of result = 0 before the decrement? How would that help?

Reply:
Thanks for the clarification, I have updated the code.